erigon-lib/downloader/downloader.go

```diff
@@ -2319,6 +2293,10 @@ func (d *Downloader) ReCalcStats(interval time.Duration) {
 
 	d.lock.Unlock()
 
+	if stats.Completed {
+		d.saveAllCompleteFlag()
```

Comment:
There're two steps of downloading: 1. header chain segments(headers and bodies) 2. other segments(transactions...)
stats.Completed is true after the first step completed, but will be re-written to false once step2 starts.

should we wait all types of segments complete and then set the flag?

Reply:
add some logs locally and here's logs test on sepolia:
```
Stats.Completed false
[INFO] [10-17|23:03:55.301] [1/6 OtterSync] Downloading              progress="(228/228 files) 99.71% - 2.7GB/2.7GB" rate=5.9MB/s time-left=0hrs:0m total-time=1m40s download-rate=5.9MB/s completion-rate=5.9MB/s alloc=552.7MB sys=653.8MB
Stats.Completed true
header and body snapshots downloaded completely
other snapshots downloading started
[INFO] [10-17|23:03:55.810] [snapshots] no metadata yet              files=90 list=domain/v1-accounts.0-128.bt,domain/v1-receipt.200-202.kv,v1-006710-006720-transactions.idx,v1-006730-006740-transactions.seg,domain/v1-accounts.192-200.bt,...
Stats.Completed false
```

Reply:
You right. Maybe i must use “StageSnapshots progress > 0”